From cd438ebc524bc27b6200c70ccb6ed9f8d0271a10 Mon Sep 17 00:00:00 2001
From: Hui Peng <phui@google.com>
Date: Wed, 19 Jul 2023 18:14:59 +0000
Subject: [PATCH] Revert "Fix a type confusion bug in bta_av_setconfig_rej"

This reverts commit bbd88e88ce749aab87178e189a05e5a356d0631c.

Reason for revert: b/281788858
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:c3356f17866099405ef6c01244211af4bd93db2c)
Merged-In: I0f45c87ba5f0b2c84843e568ca117439b42d1ed3
Change-Id: I0f45c87ba5f0b2c84843e568ca117439b42d1ed3
---
 system/bta/av/bta_av_aact.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/system/bta/av/bta_av_aact.cc b/system/bta/av/bta_av_aact.cc
index 39f95e5040..d0db36ea3e 100644
--- a/system/bta/av/bta_av_aact.cc
+++ b/system/bta/av/bta_av_aact.cc
@@ -1740,14 +1740,14 @@ void bta_av_getcap_results(tBTA_AV_SCB* p_scb, tBTA_AV_DATA* p_data) {
  ******************************************************************************/
 void bta_av_setconfig_rej(tBTA_AV_SCB* p_scb, tBTA_AV_DATA* p_data) {
   tBTA_AV_REJECT reject;
+  uint8_t avdt_handle = p_data->ci_setconfig.avdt_handle;
 
-  bta_av_adjust_seps_idx(p_scb, p_scb->avdt_handle);
-
+  bta_av_adjust_seps_idx(p_scb, avdt_handle);
   LOG_INFO("%s: sep_idx=%d avdt_handle=%d bta_handle=0x%x", __func__,
            p_scb->sep_idx, p_scb->avdt_handle, p_scb->hndl);
   AVDT_ConfigRsp(p_scb->avdt_handle, p_scb->avdt_label, AVDT_ERR_UNSUP_CFG, 0);
 
-  reject.bd_addr = p_scb->PeerAddress();
+  reject.bd_addr = p_data->str_msg.bd_addr;
   reject.hndl = p_scb->hndl;
 
   tBTA_AV bta_av_data;
